sample-plugin-candlestick-chart-new/src/CandlestickChart.js

```diff
@@ -0,0 +1,153 @@
+			.merge((d, c) => {d.ema50 = c;})
+			.accessor(d => d.ema50);
+
+		const slowSTO = stochasticOscillator()
+			.options({ windowSize: 14, kWindowSize: 3 })
+			.merge((d, c) => {d.slowSTO = c;})
+			.accessor(d => d.slowSTO);
+
+		const fastSTO = stochasticOscillator()
+			.options({ windowSize: 14, kWindowSize: 1 })
+			.merge((d, c) => {d.fastSTO = c;})
+			.accessor(d => d.fastSTO);
+
+		const fullSTO = stochasticOscillator()
+			.options({ windowSize: 14, kWindowSize: 3, dWindowSize: 4 })
+			.merge((d, c) => {d.fullSTO = c;})
+			.accessor(d => d.fullSTO);
+
+		const bb = bollingerBand()
+			.merge((d, c) => {d.bb = c;})
+			.accessor(d => d.bb);
+		// END helper functions for additional feature support
+
+		const calculatedData = bb(ema20(ema50(slowSTO(fastSTO(fullSTO(initialData))))));
+		const xScaleProvider = discontinuousTimeScaleProvider.inputDateAccessor(d => d.date);
+
+		const {
+			data,
+			xScale,
+			xAccessor,
+			displayXAccessor,
+		} = xScaleProvider(calculatedData);
+
+		const start = xAccessor(last(data));
+		const end = xAccessor(data[Math.max(0, data.length - 150)]);
+		const xExtents = [start, end];
+
+		return (
+			<ChartCanvas 
+				height={height}
+				width={width}
+				ratio={ratio}
+				margin={margin}
+				seriesName={'StockChart'}
+				type={type}
+				data={data}
+				xScale={xScale}
+				xAccessor={xAccessor}
+				displayXAccessor={displayXAccessor}
+				xExtents={xExtents}
+			>
+				<Chart 
+					id={1} 
+					height={450}
+					yExtents={[d => [d.high, d.low], bb.accessor(), ema20.accessor(), ema50.accessor()]}
+					padding={{ top: 10, bottom: 20 }}
+				>
+					<YAxis 
+						axisAt="right" 
+						orient="right" 
+						ticks={10} 
+						{...yGrid} 
+						inverted={true}
+						tickStroke="#0F0F0F" 
+					/>
+					<XAxis 
+						axisAt="bottom" 
+						orient="bottom"
+						{...xGrid}
+						tickStroke={'#0F0F0F'}
+						stroke={'#0F0F0F'}
+					/>
+					<MouseCoordinateY
+						at="right"
+						orient="right"
+						displayFormat={format(".2f")} 
+					/>
+					<CandlestickSeries
+						stroke={d => d.close > d.open ? "#6BA583" : "#DB0000"}
+						wickStroke={d => d.close > d.open ? "#6BA583" : "#DB0000"}
+						fill={d => d.close > d.open ? "#6BA583" : "#DB0000"} 
+					/>
+					<EdgeIndicator 
+						itemType="last" 
+						orient="right" 
+						edgeAt="right"
+						yAccessor={d => d.close} 
+						fill={d => d.close > d.open ? "#6BA583" : "#DB0000"}
+					/>
+					<OHLCTooltip 
+						origin={[-40, -10]}
+						onClick={e => console.log(e)}
+					/>
+					<CrossHairCursor stroke="#FFFFFF" />
+				</Chart>
+			</ChartCanvas>
+		);
+	}
+}
+CandlestickChart.propTypes = {
+	data: PropTypes.array.isRequired,
+	width: PropTypes.number.isRequired,
+	ratio: PropTypes.number.isRequired,
+	type: PropTypes.oneOf(["svg", "hybrid"]).isRequired,
+};
+
+CandlestickChart.defaultProps = {
+	type: "svg",
+};
+
+CandlestickChart = fitWidth(CandlestickChart);
+
+export default CandlestickChart;
```

Comment:
nit: newline

Reply:
Hey Pearce, thanks again for the comprehensive review-- I'm going to push to get this done by EoW

Just to clarify, does this comment just mean to add a new blank line after the end?